core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -313,6 +303,33 @@ final class ShuffleBlockFetcherIterator(
   }
 }
 
+// Helper class that ensures a ManagedBuffer is released on InputStream.close()
+private class WrappedInputStream(
+              delegate: InputStream, buf: ManagedBuffer, var currentResult: FetchResult)
```

Comment:
Why do we need to pass `currentResult` here?  It's not obvious on first read and I'm worried that someone might remove it by accident unless we add a comment explaining why we need to keep it.